packages/chain-adapters/src/dogecoin/DogecoinChainAdapter.ts

```diff
@@ -0,0 +1,458 @@
+      }
+
+      const path = toRootDerivationPath(bip44Params)
+      const pubkey = await this.getPublicKey(wallet, bip44Params, accountType)
+      console.info(`pubkey: ${pubkey.xpub}`)
+      const { data: utxos } = await this.providers.http.getUtxos({
+        pubkey: pubkey.xpub
+      })
+
+      if (!supportsBTC(wallet))
+        throw new Error(
+          'DogecoinChainAdapter: signTransaction wallet does not support signing btc txs'
+        )
+
+      const account = await this.getAccount(pubkey.xpub)
+
+      const coinSelectResult = utxoSelect({
+        utxos,
+        to,
+        satoshiPerByte,
+        sendMax,
+        value,
+        opReturnData
+      })
+
+      if (!coinSelectResult || !coinSelectResult.inputs || !coinSelectResult.outputs) {
+        throw new Error(`DogecoinChainAdapter: coinSelect didn't select coins`)
+      }
+
+      const { inputs, outputs } = coinSelectResult
+
+      const signTxInputs: BTCSignTxInput[] = []
+      for (const input of inputs) {
+        if (!input.path) continue
+        const getTransactionResponse = await this.providers.http.getTransaction({
+          txid: input.txid
+        })
+        const inputTx = getTransactionResponse.data
+
+        signTxInputs.push({
+          addressNList: bip32ToAddressNList(input.path),
+          // https://github.com/shapeshift/hdwallet/issues/362
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          scriptType: accountTypeToScriptType[accountType] as any,
+          amount: String(input.value),
+          vout: input.vout,
+          txid: input.txid,
+          hex: inputTx.hex
+        })
+      }
+
+      const signTxOutputs: BTCSignTxOutput[] = outputs.map((out) => {
+        const amount = String(out.value)
+        if (!out.address) {
+          return {
+            addressType: BTCOutputAddressType.Change,
+            amount,
+            addressNList: bip32ToAddressNList(
+              `${path}/1/${String(account.chainSpecific.nextChangeAddressIndex)}`
+            ),
+            scriptType: accountTypeToOutputScriptType[accountType],
+            isChange: true
+          }
+        } else {
+          return {
+            addressType: BTCOutputAddressType.Spend,
+            amount,
+            address: out.address
+          }
+        }
+      })
+
+      const txToSign: BTCSignTx = {
+        coin: this.coinName,
+        inputs: signTxInputs,
+        outputs: signTxOutputs,
+        opReturnData
+      }
+      return { txToSign }
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  buildBIP44Params(params: Partial<BIP44Params>): BIP44Params {
+    return { ...ChainAdapter.defaultBIP44Params, ...params }
+  }
+
+  async signTransaction(
+    signTxInput: SignTxInput<ChainTxType<KnownChainIds.DogecoinMainnet>>
+  ): Promise<string> {
+    try {
+      const { txToSign, wallet } = signTxInput
+      if (!supportsBTC(wallet))
```

Comment:
do we have/need a `supportsDoge` fn?

Reply:
no. "supports" here is whether the wallet knows how to do bitcoin things, which is the same as doing dogecoin things. there's no dogecoin HDWallet impl, we just reference it as coinType 3 and with coinName `dogecoin`